.github/workflows/tests.yml

```diff
@@ -0,0 +1,82 @@
+name: Tests
+
+on:
+  push:
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  check_duplicate_runs:
+    name: Check for duplicate runs
+    continue-on-error: true
+    runs-on: ubuntu-latest
+    outputs:
+      should_skip: ${{ steps.skip_check.outputs.should_skip }}
+    steps:
+      - id: skip_check
+        uses: fkirc/skip-duplicate-actions@master
+        with:
+          concurrent_skipping: always
+          cancel_others: true
+          skip_after_successful_duplicate: true
+          paths_ignore: '["**/README.md", "**/CHANGELOG.md", "**/LICENSE.md"]'
+          do_not_skip: '["pull_request"]'
+
+  tests:
+    name: Run tests (Python ${{matrix.python}})
+    needs: check_duplicate_runs
+    if: ${{ needs.check_duplicate_runs.outputs.should_skip != 'true' }}
+
+    strategy:
+      matrix:
+        python:
```

Comment:
I've removed python3.5 from text matrix because `jupyter_client` dependency does not support it:
https://travis-ci.org/github/jupyter/kernel_gateway/jobs/766251855
```
from jupyter_client.channels import HBChannel
  File "/home/travis/virtualenv/python3.5.10/lib/python3.5/site-packages/jupyter_client/channels.py", line 47
    time_to_dead: float = 1.0
                ^
SyntaxError: invalid syntax
```

Also I've added python3.9 to the matrix because it was released recently.